src/Microsoft.ML.TorchSharp/NasBert/TextClassificationTrainer.cs

```diff
@@ -812,27 +842,83 @@ private sealed class Mapper : MapperBase
+                meta.Add(AnnotationUtils.Kinds.TrainingLabelValues, keyType, getter);
+                meta.Add(AnnotationUtils.Kinds.SlotNames, keyType, getter);
 
-                info[0] = new DataViewSchema.DetachedColumn(_parent._options.OutputColumnName, new KeyDataViewType(typeof(uint), _parent._options.NumberOfClasses), _labelCol.Annotations);
+                info[0] = new DataViewSchema.DetachedColumn(_parent._options.PredictionColumnName, new KeyDataViewType(typeof(uint), _parent._options.NumberOfClasses), _labelCol.Annotations);
+
+                info[1] = new DataViewSchema.DetachedColumn(_parent._options.ScoreColumnName, new VectorDataViewType(NumberDataViewType.Single, _parent._options.NumberOfClasses), meta.ToAnnotations());
                 return info;
             }
 
+            private Delegate GetLabelAnnotations<T>()
+            {
+                return _labelCol.Annotations.GetGetter<VBuffer<T>>(_labelCol.Annotations.Schema[AnnotationUtils.Kinds.KeyValues]);
+            }
+
+            private ValueGetter<uint> GetScoreColumnSetId(DataViewSchema schema)
+            {
+                int c;
+                var max = schema.GetMaxAnnotationKind(out c, AnnotationUtils.Kinds.ScoreColumnSetId);
+                uint id = checked(max + 1);
+                return
+                    (ref uint dst) => dst = id;
+            }
+
             protected override Delegate MakeGetter(DataViewRow input, int iinfo, Func<int, bool> activeOutput, out Action disposer)
+                => throw new NotImplementedException("This should never be called!");
+
+            private Delegate CreateGetter(DataViewRow input, int iinfo, TensorCacher outputCacher)
             {
-                Host.AssertValue(input);
                 var ch = Host.Start("Make Getter");
-                disposer = null;
+                if (iinfo == 0)
+                    return MakePredictedLabelGetter(input, ch, outputCacher);
+                else
+                    return MakeScoreGetter(input, ch, outputCacher);
+            }
+
+            public override Delegate[] CreateGetters(DataViewRow input, Func<int, bool> activeOutput, out Action disposer)
+            {
+                Host.AssertValue(input);
+                Contracts.Assert(input.Schema == InputSchema);
 
-                Host.Assert(_inputColIndices.All(i => input.IsColumnActive(input.Schema[i])));
+                TensorCacher outputCacher = new TensorCacher();
+                var ch = Host.Start("Make Getters");
+                _parent._model.eval();
 
+                int n = OutputColumns.Value.Length;
+                var result = new Delegate[n];
+                for (int i = 0; i < n; i++)
+                {
+                    if (!activeOutput(i))
+                        continue;
```

Comment:
>continue [](http://example.com/codeflow?start=24&length=8)

Is it allowed to return nulls inside the result array? just wondering :-)

Reply:
I believe so. I actually took that exact code from another transformer that has been around for a while so it should allow nulls. And we have tests that cause nulls to be populated and they pass fine so we should be only using the output after checking its not null.